src/Channels.Networking.Windows.RIO/Internal/RioThread.cs

```diff
@@ -120,7 +126,10 @@ private static void OnThreadStart(object state)
                             RioTcpConnection connection;
                             if (thread._connections.TryGetValue(result.ConnectionCorrelation, out connection))
                             {
-                                connection.Complete(result.Status, result.RequestCorrelation, result.BytesTransferred);
+                                ThreadPool.QueueUserWorkItem(
```

Comment:
Yea, I know it does for all of the implementations 😄 . It would be better to let the server code dispatch than this low level code.